### PR TITLE
Competitive programming : Merge Sort Documentation

### PR DESCRIPTION
**Pull request by ksheetiz**
<hr>

## Description 📜

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. 
Fixes # <2884> 

<hr>

## Type of change 📝

- [ ] Au#dio (Should be in mp3 format Includes speech clarity, Concise ,Low distortion)
- [ ] Vi#deo (Animations, screen-recordings, presentations and regular explanatory films are all possibilties etc)
- [x] Doc#umentation (Content Creation in the form of codes or tutorials)
- [ ] Other (If you choose other, Please mention changes below) 

<hr>

## Domain of Contribution 📊

- [ ] Android Dev #(Flutter)
- [ ] Android Dev #(Java)
- [ ] Android Dev #(Kotlin)
- [ ] Backend Dev #(Java)
- [ ] Backend Dev #(.NET)
- [ ] Backend Dev #(PHP)
- [ ] Backend Dev #(Python)
- [ ] C/#CPP
- [x] Competitive #Programming
- [ ] Cyber #Security
- [ ] DS#A
- [ ] Data#base
- [ ] Datascience with #Python
- [ ] Datascience with #R
- [ ] Frontend Dev #HTML/CSS/JS
- [ ] Frontend Dev #React/Angular/Vue
- [ ] Go#lang
- [ ] Interview #Prep
- [ ] Java_#Domain
- [ ] Java#Script
- [ ] ME#RN
- [ ] Machine #Learning
- [ ] Open #Source
- [ ] Python_#Domain
- [ ] Ru#st
- [ ] Statis#tics
- [ ] UI/#UX

<hr>

## Checklist ✅

- [x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-con
- tributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x] I have performed a self-review of my own code or work.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generates no new warnings.
- [x] I'm GWOC'21 contributor

<hr>

## Screenshots / Gif (Optional) 📸

<hr>

### CODE REVIEW

**Comment by ksheetiz** [comment]
please look into my PR @Suvraneel

**Comment by Suvraneel** [comment]
@ksheetiz  I'm not from the competitive programming domain, kindly tag correct mentors

**Comment by ksheetiz** [comment]
please look into my PR @Rishabh510

File: Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+MergeSort(arr[], l,  r) <br/>
+If r > l <br/>
+     1. Find the middle point to divide the array into two halves: <br/>  
+             middle m = l+ (r-l)/2 <br/>
+     2. Call mergeSort for first half:   <br/>
+             Call mergeSort(arr, l, m) <br/>
+     3. Call mergeSort for second half: <br/>
+             Call mergeSort(arr, m+1, r) <br/>
+     4. Merge the two halves sorted in step 2 and 3: <br/>
+             Call merge(arr, l, m, r) <br/>
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md:15]: Please add this points via bullet points, make sure you preview it.

File: Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+### Complexity analysis.
+Sorting arrays on different machines. Merge Sort is a recursive algorithm and time complexity can be expressed as following recurrence relation. 
+T(n) = 2T(n/2) + θ(n)
+
+Auxiliary Space: O(n) <br/>
+Algorithmic Paradigm: Divide and Conquer <br/>
+Sorting In Place: No in a typical implementation <br/>
+Stable: Yes <br/>
+
+The best-case and average-case complexity of the quick sort algorithm is O(n logn).
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md:101]: Highlight keywords wherever required.

File: Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+
+    int s = 0;
+    int e = arr.size()-1;
+    mergesort(arr,s,e);
+    for(int i : arr){
+        cout<<i<<" ,";
+    }
+}
+```
+
```
Comment (ravireddy07) [inline Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md:90]: Add `input/output` analysis as well

**Review by ravireddy07 — CHANGES_REQUESTED** [review]

File: Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md (hunk 3 above)
Comment (ksheetiz) [reply Competitive_Programming/C++/Searching & Sorting/Merge_Sort_Implementation_and_analysis.md:101]: thanks @ravireddy07 for pointing out the mistakes. I have made the respective changes please see to it @ravireddy07

**Comment by ravireddy07** [comment]
why you deleted that `README` file @ksheetiz, That's inner roadmap file which gone be filled later by Supervisor's.
Please put it back.

**Comment by ksheetiz** [comment]
@ravireddy07 i have putted it back please review it Sorry for the mistake

**Review by ravireddy07 — APPROVED** [review]
Please update the status of the PR in PR Tracker sheet.

**Comment by ksheetiz** [comment]
@ravireddy07 that i think has to be done by Mentor i guess and thanks for reviewing it

**Comment by ravireddy07** [comment]
Supervisor will merge approved PR's only. NM, i'll update

**Pull request merged**